tests/unit/state-mixin-test.js

```diff
@@ -104,3 +106,16 @@ test('teardownStateFor properly clears a state for a key', function(assert) {
   assert.equal(subject.get('state.firstName'), undefined);
   assert.ok(!subject.service.hasStateFor('person2'));
 });
+
+test('Creating a state service with an unknown StateFactory throws error', function(assert) {
+  assert.expect(1);
+  
+  Service = Ember.Service.extend(StateMixin, {
+    stateName: 'test-state-that-does-not-exist',
+    container: container
+  });
+
+  assert.throws(function() {
+    Service.create();
+  }, TypeError, 'Unknown StateFactory: `test-state-that-does-not-exist`');
```

Comment:
It's global as part of every JS env. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError http://www.ecma-international.org/ecma-262/5.1/#sec-15.1.4.14